x/valset-pref/msg_server_test.go

```diff
@@ -113,54 +151,69 @@ func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
 			msgServer := valPref.NewMsgServerImpl(suite.App.ValidatorSetPreferenceKeeper)
 			c := sdk.WrapSDKContext(suite.Ctx)
 
+			if test.setExistingDelegations {
+				amountToFund := sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 100_000_000)} // 100 osmo
+				suite.FundAcc(test.delegator, amountToFund)
+
+				err := suite.PrepareExistingDelegations(suite.Ctx, valAddrs, test.delegator, test.amountToDelegate.Amount)
+				suite.Require().NoError(err)
+			}
+
 			// call the create validator set preference
 			_, err := msgServer.SetValidatorSetPreference(c, types.NewMsgSetValidatorSetPreference(test.delegator, test.preferences))
 			if test.expectPass {
 				suite.Require().NoError(err)
 			} else {
 				suite.Require().Error(err)
 			}
-
 		})
 	}
 }
 
 func (suite *KeeperTestSuite) TestDelegateToValidatorSet() {
 	suite.SetupTest()
 
-	// prepare validators to delegate to
-	preferences := suite.PrepareDelegateToValidatorSet()
-
-	amountToFund := sdk.Coins{sdk.NewInt64Coin(sdk.DefaultBondDenom, 100_000_000)} // 100 osmo
+	// valset test setup
+	valAddrs, preferences, amountToFund := suite.SetupValidatorsAndDelegations()
 
 	tests := []struct {
-		name           string
-		delegator      sdk.AccAddress
-		coin           sdk.Coin  // amount to delegate
-		expectedShares []sdk.Dec // expected shares after delegation
-		expectPass     bool
-		valSetExists   bool
+		name                   string
+		delegator              sdk.AccAddress
+		amountToDelegate       sdk.Coin  // amount to delegate
+		expectedShares         []sdk.Dec // expected shares after delegation
+		setExistingDelegations bool
+		setValSet              bool
+		expectPass             bool
 	}{
 		{
-			name:           "Delegate to valid validators",
-			delegator:      sdk.AccAddress([]byte("addr1---------------")),
-			coin:           sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(10_000_000)),
-			expectedShares: []sdk.Dec{sdk.NewDec(2_000_000), sdk.NewDec(3_300_000), sdk.NewDec(1_200_000), sdk.NewDec(3_500_000)},
-			expectPass:     true,
+			name:             "Delegate to valid validators",
+			delegator:        sdk.AccAddress([]byte("addr1---------------")),
+			amountToDelegate: sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(10_000_000)),
+			setValSet:        true,
+			expectedShares:   []sdk.Dec{sdk.NewDec(2_000_000), sdk.NewDec(3_300_000), sdk.NewDec(1_200_000), sdk.NewDec(3_500_000)},
+			expectPass:       true,
```

Comment:
maybe add in title something about creating a val set then delegating to that set (referring to the setValSet input option)